src/tests/ftest/control/dmg_storage_query.py

```diff
@@ -40,155 +39,149 @@ class DmgStorageQuery(TestWithServers):
-
-        # Parse stdout into a dict check that devices information is displayed.
-        smd_info = query_smd_info(dmg_query_smd.stdout)
-        self.check_smd_out(smd_info, "devices", "pools")
+        # Get the storage device information, parse and check devices info
+        devs_info = self.get_device_info()
 
-    def test_dmg_storage_query_smd_pools(self):
+        # Check if the number of devices match the config
+        msg = "Number of devs do not match cfg: {}".format(len(self.bdev_list))
+        self.assertEqual(len(self.bdev_list), len(devs_info.values()[0]), msg)
+
+        # Check that number of targets match the config
+        targets = 0
+        for devs in devs_info.values()[0]:
+            targets = len(devs[1].split(" "))
+        if self.targets != len(targets):
+            self.fail("Wrong number of targets found: {}".format(targets))
+
+    @avocado.fail_on(CommandFailure)
+    def test_dmg_storage_query_pools(self):
         """
         JIRA ID: DAOS-3925
 
-        Test Description: Test 'dmg storage query smd -p' command.
+        Test Description: Test 'dmg storage query list-pools' command.
 
-        :avocado: tags=all,pr,hw,small,dmg_storage_query,smd_pools,basic
+        :avocado: tags=all,pr,hw,small,storage_query_pools,basic,dmg
         """
-        # Create pool and get the storage smd information
-        self.prepare_pool(self.dmg)
-        try:
-            dmg_query_smd = self.dmg.storage_query_smd(pools=False)
-        except CommandFailure as err:
-            self.fail("Failed running storage query command: {}".format(err))
-
-        # Parse stdout into a dict and check pools information is displayed.
-        smd_info = query_smd_info(dmg_query_smd.stdout)
-        self.check_smd_out(smd_info, "pools", "devices")
-
-        # Destroy pool and get the storage smd information
-        self.pool.destroy()
-        try:
-            dmg_query_smd = self.dmg.storage_query_smd(pools=False)
-        except CommandFailure as err:
-            self.fail("Failed running storage query command: {}".format(err))
+        # Create pool and get the storage smd information, then verfify info
+        self.prepare_pool()
+        pools_info = self.get_pool_info(verbose=True)
 
-        # Check that only devices information is displayed.
-        smd_info = query_smd_info(dmg_query_smd.stdout)
-        self.check_smd_out(smd_info, "devices", "pools")
+        # Check pool uuid
+        for pool in pools_info.values()[0]:
+            self.assertEqual(self.pool.pool.get_uuid_str(), pool[0].upper())
 
-    def test_dmg_storage_query_blobstore_health(self):
+        # Destroy pool and get pool information and check there is no pool
+        self.pool.destroy()
+        no_pool_info = self.get_pool_info()
+        self.assertFalse(no_pool_info, "No pools should be detected.")
+
+        # Check that number of pool blobs match the number of targets
```

Comment:
I'm not familiar with vos targets and blobs, but isn't it better to do this step before destroying the pool?

Reply:
Yes, that's a good point. I'll move this check to the bottom.